src/recode.jl

```diff
@@ -49,16 +49,32 @@ A user defined type could override this method to define an appropriate test fun
 @inline recode_in(x, collection::Set) = x in collection
 @inline recode_in(x, collection) = any(x ≅ y for y in collection)
 
+
+function optimize_pairs(pairs::Tuple{Vararg{Pair}})::Vector{Pair}
+    pair_vec = Vector{Pair}(undef, length(pairs))
+    for (ind_pair, p) in enumerate(pairs)
+        if typeof(p.first) <: AbstractArray
+            pair_vec[ind_pair] = Set(p.first) => p.second
+        else
+            pair_vec[ind_pair] = p
+        end
+    end
+    return pair_vec
+end
```

Comment:
It would be more logical and possibly more efficient to only define an `optimize_pair` function operating on a single value at a time, and do `opt_pairs = map(optimize_pair, pairs)`. The point is to get a tuple of concrete types rather than a vector of abstract types.